src/openrct2/scenario/ScenarioRepository.cpp

```diff
@@ -612,7 +612,7 @@ class ScenarioRepository final : public IScenarioRepository
                 scenario_highscore_entry* highscore = InsertHighscore();
                 highscore->fileName = fs.ReadString();
                 highscore->name = fs.ReadString();
-                highscore->company_value = fs.ReadValue<money32>();
+                highscore->company_value = fileVersion == 1 ? fs.ReadValue<money32>() : fs.ReadValue<money64>();
```

Comment:
doesnt the money32 need to be converted? Is a money32_undefined possible or `COMPANY_VALUE_ON_FAILED_OBJECTIVE`

Reply:
No, these are not possible. Only actual company values end up in the highscores.